### PR TITLE
Update cloudwatch exporter dependency

### PR DESCRIPTION
**Pull request by andriikushch**
#### PR Description

This PR adds a new configuration parameter for the Cloudwatch exporter. It allows the specification of the AWS SDK version to be used. `v1` is the default version. 

It was achieved by updating and integrating the version of the `github.com/nerdswords/yet-another-cloudwatch-exporter` package and all dependencies that it depends on.

According to the https://github.com/nerdswords/yet-another-cloudwatch-exporter/blob/a275f94e5d33b3b5330ecfced86c290e1db05928/pkg/config/feature_flags.go#L10 , using the AWS SDK v2, should improve performance.

# Breaking change

After this change, using aliases for the AWS service names is no longer allowed.

#### Which issue(s) this PR fixes

N.A.

A related change in the Alloy project:
- https://github.com/grafana/alloy/pull/839

#### Notes to the Reviewer

There is a breaking change in the PR.

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

**Comment by andriikushch** [comment]
Related :

- https://github.com/grafana/postgres_exporter/pull/18

File: docs/sources/flow/reference/components/prometheus.exporter.cloudwatch.md [hunk 1]
```diff
@@ -124,11 +126,12 @@ prometheus.exporter.cloudwatch "queues" {
...
+| ------------------------- | ------------------- |----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|----------|
+| `sts_region`              | `string`            | AWS region to use when calling [STS][] for retrieving account information.                                                                                                                                                             |         | yes      |
+| `aws_sdk_version`         | `string`            | AWS SDK version                                                                                                                                                                                                                        | `v2`    | no       |
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.cloudwatch.md:132]:
```suggestion
| `aws_sdk_version`         | `string`            | AWS SDK version.                                                                                                                                                                                                                        | `v2`    | no       |
```

**Comment by clayton-cornell** [comment]
Doc changes look OK to me. Over to the @grafana/grafana-agent-maintainers for review.

**Comment by andriikushch** [comment]
A similar change was made in the [Alloy](https://github.com/grafana/alloy) project. It should be used in case these changes are required.

**Pull request closed without merging**